arch/arm/core/aarch32/mpu/arm_mpu.c

```diff
@@ -107,8 +83,13 @@ static int mpu_configure_region(const uint8_t index,
 
 	/* Populate internal ARM MPU region configuration structure. */
 	region_conf.base = new_region->start;
+#if defined(CONFIG_CPU_CORTEX_R)
+	region_conf.size = size_to_mpu_rasr_size(new_region->size);
+	region_conf.attr.rasr = new_region->attr.rasr_attr;
+#else
```

Comment:
Below get_region_attr_from_k_mem_partition_info seems to abstract from the MPU region attribute details (e.g. v7 or v8), any specific reason to handle cortex R differently with an #ifdef here instead?

Reply:
no reason. I updated it so that attr is abstracted under get_region_attr_from_k_mem_partition_info